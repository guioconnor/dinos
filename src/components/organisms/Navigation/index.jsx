import React from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";

const Nav = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 100;
  padding: 10px;
  background: #122;

  & a {
    display: inline-block;
    margin: 0 10px;
    padding: 10px;
    border-radius: 3px;
    text-decoration: none;
    color: #cde;
    transition: 0.3s background;
  }

  & a.active,
  & a:hover {
    background: #567;
  }
`;

const Navigation = () => (
  <Nav>
    <NavLink exact activeClassName="active" to="/dinos">
      Dinos
    </NavLink>
    <NavLink activeClassName="active" to={{ pathname: "/dinos/memory" }}>
      Dino Memory game
    </NavLink>
    <NavLink activeClassName="active" to={{ pathname: "/animals/memory" }}>
      Animals Memory game
    </NavLink>
    <NavLink activeClassName="active" to={{ pathname: "/birds/memory" }}>
      Birds Memory game
    </NavLink>
  </Nav>
);

export default Navigation;
