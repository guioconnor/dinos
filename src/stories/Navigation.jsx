import React from "react";
import { storiesOf } from "@storybook/react";
import Navigation from "../components/organisms/Navigation";

storiesOf("Navigation", module).add("default", Navigation);
