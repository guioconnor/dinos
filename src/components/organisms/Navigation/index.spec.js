import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Navigation from "./index";

describe("<Navigation />", () => {
  const wrapper = shallow(<Navigation dinoData={[]} />);

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
