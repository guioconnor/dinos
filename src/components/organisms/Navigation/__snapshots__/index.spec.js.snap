// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render correctly 1`] = `
<styled.nav>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    exact={true}
    to="/dinos"
  >
    Dinos
  </NavLink>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to={
      Object {
        "pathname": "/dinos/memory",
      }
    }
  >
    Dino Memory game
  </NavLink>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to={
      Object {
        "pathname": "/animals/memory",
      }
    }
  >
    Animals Memory game
  </NavLink>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to={
      Object {
        "pathname": "/birds/memory",
      }
    }
  >
    Birds Memory game
  </NavLink>
</styled.nav>
`;
